modules/caddyhttp/reverseproxy/fastcgi/client.go

```diff
@@ -442,9 +459,19 @@ func (c *FCGIClient) Request(p map[string]string, req io.Reader) (resp *http.Res
 	resp.ContentLength, _ = strconv.ParseInt(resp.Header.Get("Content-Length"), 10, 64)
 
 	if chunked(resp.TransferEncoding) {
-		resp.Body = clientCloser{c, httputil.NewChunkedReader(rb)}
+		resp.Body = clientCloser{
+			FCGIClient: c,
+			Reader:     httputil.NewChunkedReader(rb),
+			status:     resp.StatusCode,
+			logger:     c.logger,
+		}
 	} else {
-		resp.Body = clientCloser{c, io.NopCloser(rb)}
+		resp.Body = clientCloser{
+			FCGIClient: c,
+			Reader:     rb,
+			status:     resp.StatusCode,
+			logger:     c.logger,
+		}
 	}
```

Comment:
Maybe we could refactor this to create the `Reader` as a variable in an `if/else` then just make the `clientCloser` once with the reader. A bit less duplication of code.

Reply:
Yeah that's what I was alluding to when I mentioned
> The changes could be a little cleaner (especially the handling of transfer encoding)

That is to say, this was done on purpose so that we keep the symmetry with the old code, which should make the merge choice a little more obvious.

Should we just do it the right way now?

Reply:
/cc @WeidiDeng up to you I'd say.

I'm not sure what order we'll end up merging, but I figure either of you can resolve it afterwards.

@WeidiDeng how much work do you have left on your branch? Do you think it's pretty much good to go? Any loose threads?

Reply:
@francislavoie There is not much to be done at the moment, except if there are bug reports. Fastcgi has one last optimization that I can think of is reusing connections, but that won't be for a while.

Reply:
Alright. I guess we can merge this one first and then you can rebase & fixup your branch @WeidiDeng